server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2655,8 +2655,7 @@ private static void persistMetadata(
 
 
     private DocumentMapperForType docMapper() {
-        return mapperService.documentMapperWithAutoCreate(
-            mapperService.resolveDocumentType(MapperService.SINGLE_MAPPING_NAME));
+        return mapperService.documentMapperWithAutoCreate();
```

Comment:
To check my understanding, this is fine to change because the type is no longer relevant to indexing (for example we already removed type information from the translog)?

Reply:
Correct - we can't remove `MapperService#resolveDocumentType` yet because it's still used by `DocumentMapperParser`, but that's on the list...